src/Integration.Vsix/SonarLintTracker/IssuesSnapshot.cs

```diff
@@ -0,0 +1,140 @@
+        }
+
+        public override int Count => this.issueMarkers.Count;
+
+        public override int VersionNumber => this.versionNumber;
+
+        public override bool TryGetValue(int index, string columnName, out object content)
+        {
+            if (index < 0 || this.issueMarkers.Count <= index)
+            {
+                content = null;
+                return false;
+            }
+
+            // TODO use correct issue type, such as bug, vulnerability, code smell
+            if (columnName == StandardTableKeyNames.DocumentName)
+            {
+                content = filePath;
+                return true;
+            }
+
+            if (columnName == StandardTableKeyNames.Line)
+            {
+                content = this.issueMarkers[index].Span.Start.GetContainingLine().LineNumber;
+                return true;
+            }
+
+            if (columnName == StandardTableKeyNames.Column)
+            {
+                var position = this.issueMarkers[index].Span.Start;
+                var line = position.GetContainingLine();
+                content = position.Position - line.Start.Position;
+                return true;
+            }
+
+            if (columnName == StandardTableKeyNames.Text)
+            {
+                content = this.issueMarkers[index].Issue.Message;
+                return true;
+            }
+
+            if (columnName == StandardTableKeyNames.ErrorSeverity)
+            {
+                // TODO use correct icon type depending on severity
+                content = __VSERRORCATEGORY.EC_WARNING;
+                return true;
+            }
+
+            if (columnName == StandardTableKeyNames.BuildTool)
+            {
+                // TODO get correct analyzer name
+                content = "SonarJS [SonarLint for Visual Studio 2015]";
+                return true;
+            }
+
+            if (columnName == StandardTableKeyNames.ErrorCode)
+            {
+                content = this.issueMarkers[index].Issue.RuleKey;
+                return true;
+            }
+
+            if ((columnName == StandardTableKeyNames.ErrorCodeToolTip) || (columnName == StandardTableKeyNames.HelpLink))
+            {
+                // TODO use correct version
+                // TODO add JavaScript rules on website
+                //content = string.Format(CultureInfo.InvariantCulture, "http://www.sonarlint.org/visualstudio/rules/index.html#version=5.9.0.992&ruleId={0}", this.issueMarkers[index].Issue.RuleKey);
+                //return true;
+            }
+
+            if (columnName == StandardTableKeyNames.ProjectName)
+            {
+                // TODO get project name
+                content = "TODO";
+                return true;
+            }
+
+            content = null;
+            return false;
+        }
+
+        public override bool CanCreateDetailsContent(int index)
+        {
+            // TODO flip to true when detailed description is ready
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)